cql-parser/src/test/scala/troy/cql/parser/dml/SelectStatementParserTest.scala

```diff
@@ -323,7 +323,7 @@ class SelectStatementParserTest extends FlatSpec with Matchers {
     relations.size shouldBe 1
     relations(0).asInstanceOf[Relation.Simple].columnName shouldBe "weight"
     relations(0).asInstanceOf[Relation.Simple].operator shouldBe Operator.Equals
-    relations(0).asInstanceOf[Relation.Simple].term.asInstanceOf[FloatConstant].value shouldBe "50.4"
+    relations(0).asInstanceOf[Relation.Simple].term.asInstanceOf[FloatConstant].value shouldBe 50.4
```

Comment:
It is not a bug, this is the nature of the floats. Try executing `0.1.toFloat + 0.2` in scala console. 
Read this for more info https://www.quora.com/How-is-a-float-or-double-value-stored-in-memory-in-java